llama_index/query_engine/pandas_query_engine.py

```diff
@@ -59,19 +60,19 @@ def default_output_processor(
     try:
         tree = ast.parse(output)
         module = ast.Module(tree.body[:-1], type_ignores=[])
-        exec(ast.unparse(module), {}, local_vars)  # type: ignore
+        safe_exec(ast.unparse(module), {}, local_vars)  # type: ignore
         module_end = ast.Module(tree.body[-1:], type_ignores=[])
         module_end_str = ast.unparse(module_end)  # type: ignore
         if module_end_str.strip("'\"") != module_end_str:
             # if there's leading/trailing quotes, then we need to eval
             # string to get the actual expression
-            module_end_str = eval(module_end_str, {"np": np}, local_vars)
+            module_end_str = safe_eval(module_end_str, {"np": np}, local_vars)
         try:
             # str(pd.dataframe) will truncate output by display.max_colwidth
             # set width temporarily to extract more text
             if "max_colwidth" in output_kwargs:
                 pd.set_option("display.max_colwidth", output_kwargs["max_colwidth"])
-            output_str = str(eval(module_end_str, {"np": np}, local_vars))
+            output_str = str(safe_eval(module_end_str, {"np": np}, local_vars))
```

Comment:
I thiiiiink panadas/pd also needs to be added to the input now? Since it's not in the list of builtins? I might be wrong though, I need to test still

Reply:
will check this e2e with OpenAI API

Reply:
worked ok, although I have extended allowed imports with standard data science packages

Reply:
Thanks a ton!